runtime-transaction/src/instruction_meta.rs

```diff
@@ -0,0 +1,32 @@
+use {
+    crate::{
+        instruction_data_len::InstructionDataLenBuilder,
+        signature_details::{PrecompileSignatureDetails, PrecompileSignatureDetailsBuilder},
+    },
+    solana_pubkey::Pubkey,
+    solana_svm_transaction::instruction::SVMInstruction,
+    solana_transaction_error::TransactionError,
+};
+
+pub struct InstructionMeta {
+    pub precompile_signature_details: PrecompileSignatureDetails,
+    pub instruction_data_len: u64,
+}
+
+impl InstructionMeta {
+    pub fn try_new<'a>(
+        instructions: impl Iterator<Item = (&'a Pubkey, SVMInstruction<'a>)>,
+    ) -> Result<Self, TransactionError> {
+        let mut precompile_signature_details_builder = PrecompileSignatureDetailsBuilder::default();
+        let mut instruction_data_len_builder = InstructionDataLenBuilder::default();
```

Comment:
intent is to have more of these processor/builders.
I don't see a need rn to force them into some specific trait, some of them may return errors, while others may not.

Reply:
Also think out loud, if it's the time to think about what is the best way to maintain `InstructionMeta` to add/remove fields from it, or to support different `builder` variations. Assume this is internal structure, if change requires update all call sites, it's probably not such a big deal. Your thoughts?